### PR TITLE
NATIVE-390: Setup CI using Fastlane and Circleci

### PR DESCRIPTION
**Pull request by maxammann**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+          working_directory: android
+      - save_cache:
+          paths:
+            - ~/.gradle
+          key: gradle-{{ checksum "android/build.gradle" }}-{{ checksum "android/app/build.gradle" }}-{{ checksum "android/settings.gradle" }}
+      - run:
+          name: Fastlane Release Build
+          command: bundle exec fastlane android release
+          working_directory: android
+#      - run:
```
Comment (maxammann) [inline .circleci/config.yml]: Uncommented until the CD part

File: android/fastlane/Fastfile [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        command: "test:e2e",
+        package_path: "../package.json"
+    )
+  end
+
+  desc "experiment"
+  lane :experiment do
+    ENV["E2E_CAPS"] = 'ci_browserstack'
+    sh("echo $E2E_CAPS")
+  end
```
Comment (Schedulaar) [inline android/fastlane/Fastfile]: What's this?

File: ios/Integreat/Info.plist [hunk 3]
```diff
@@ -17,11 +17,11 @@
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>2020.3.1</string>
+	<string>0.1</string>
```
Comment (Schedulaar) [inline ios/Integreat/Info.plist]: So what's this version code?

File: version.yml [hunk 4]
```diff
@@ -0,0 +1 @@
+version_name: "0.1"
```
Comment (Schedulaar) [inline version.yml]: What's this version?

**Comment by Schedulaar** [comment]
Does circle ci build the branches or the pull request?

**Comment by maxammann** [comment]
@Schedulaar I switched to only build PRs

File: ios/Integreat/Info.plist (hunk 3 above)
Comment (maxammann) [reply ios/Integreat/Info.plist]: This is the version code which is used for builds from developers.

File: version.yml (hunk 4 above)
Comment (maxammann) [reply version.yml]: This is not defined yet properly. I will take care of this in the CD

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+          command: yarn run lint
+      - run:
+          name: Flow-type check
+          command: yarn run flow:check-now
+      - run:
+          name: Unit tests
+          command: yarn run test
+      - run:
+          name: Ruby version
+          command: ruby --version
```
[the remark this answers is not in the document]
Comment (maxammann) [reply .circleci/config.yml]: This gives useful debugging information.

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+          command: bundle check || bundle install --path vendor/bundle
+          working_directory: android
+      - save_cache:
+          key: 1-gems-{{ checksum "android/Gemfile.lock" }}-{{ arch }}
+          paths:
+            - android/vendor/bundle
+      - restore_cache:
+          name: Restore Yarn Package Cache
+          keys:
+            - 2-yarn-v1-{{ checksum "yarn.lock" }}-{{ arch }}
```
[the remark this answers is not in the document]
Comment (maxammann) [reply .circleci/config.yml]:
```
 ~ yarn --verison
yarn install v1.22.4

```

It uses yarn v1

File: .circleci/config.yml [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+          fingerprints:
+            - "9a:0f:b1:f0:97:c8:db:3f:be:87:a4:9b:09:7f:d3:af"
+      - checkout
+      - run:
+          name: "Pull Submodules"
+          command: |
+            git submodule init
+            git submodule update --remote
+      - restore_cache:
+          key: 1-gems-{{ checksum "android/Gemfile.lock" }}-{{ arch }}
```
[the remark this answers is not in the document]
Comment (maxammann) [reply .circleci/config.yml]: I can experiment with that in the CD branch, but I don't see a direct benefit. Probably safes 1-2 minutes every week.

File: .circleci/config.yml [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+      TOTAL_CPUS: 3 # For class medium+
+    shell: /bin/bash -eo pipefail
+    steps:
+      - run:
+          name: "Set CPU count properly"
+          command: |
+            echo "CPU count is ${TOTAL_CPUS}"
+      - add_ssh_keys:
+          fingerprints:
+            - "9a:0f:b1:f0:97:c8:db:3f:be:87:a4:9b:09:7f:d3:af"
```
[the remark this answers is not in the document]
Comment (maxammann) [reply .circleci/config.yml]: This is the fingerprint of the deployment key in the credentials repo.

File: .circleci/config.yml [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+      - image: circleci/android:api-28-node
+    resource_class: medium+
+    environment:
+      TOTAL_CPUS: 3 # For class medium+
+    shell: /bin/bash -eo pipefail
+    steps:
+      - run:
+          name: "Set CPU count properly"
+          command: |
+            echo "CPU count is ${TOTAL_CPUS}"
```
[the remark this answers is not in the document]
Comment (maxammann) [reply .circleci/config.yml]: This proviedes debugging info. TOTAL_CPUS is used by metro now.

**Pull request merged**